patches/chrome-android-java-src-org-chromium-chrome-browser-fullscreen-FullscreenHtmlApiHandlerBase.java.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/fullscreen/FullscreenHtmlApiHandlerBase.java b/chrome/android/java/src/org/chromium/chrome/browser/fullscreen/FullscreenHtmlApiHandlerBase.java
+index 65490bf6514e2ab2e8aa28b9c28d6310b82d1341..42fc2883b01ba6871d96c603a29b9b91466fa9c3 100644
+--- a/chrome/android/java/src/org/chromium/chrome/browser/fullscreen/FullscreenHtmlApiHandlerBase.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/fullscreen/FullscreenHtmlApiHandlerBase.java
+@@ -113,6 +113,7 @@ public abstract class FullscreenHtmlApiHandlerBase
+     @Nullable private Tab mTab;
+ 
+     private boolean mNotifyOnNextExit;
++    private boolean mTabHiddenByChangedTabs;
+ 
+     // Current ContentView. Updates when active tab is switched or WebContents is swapped
+     // in the current Tab.
+@@ -275,6 +276,7 @@ public abstract class FullscreenHtmlApiHandlerBase
+ 
+                     @Override
+                     public void onHidden(Tab tab, @TabHidingType int reason) {
++                        mTabHiddenByChangedTabs = reason == TabHidingType.CHANGED_TABS;
+                         // Clean up any fullscreen state that might impact other tabs.
+                         exitPersistentFullscreenMode();
+                     }
+@@ -443,6 +445,7 @@ public abstract class FullscreenHtmlApiHandlerBase
+ 
+     @Override
+     public void exitPersistentFullscreenMode() {
++        if (mActivity.isInPictureInPictureMode() && !mTabHiddenByChangedTabs) return;
```

Comment:
And one more quiestion, shouldn't it be apllied only for `onHidden` case? It feels like in other cases of calling `exitPersistentFullscreenMode` we shouldn't care of `mTabHiddenByChangedTabs`.

Reply:
`exitPersistentFullscreenMode` method is called also by `onActivityStateChange`, so acting only on `onHidden` case would not prevent the activity from being reset into normal mode.

<img width="2253" alt="brave-browser_–_FullscreenHtmlApiHandlerBase_java__brave-browser__all_main_" src="https://github.com/user-attachments/assets/d2addbfc-2542-4042-a1a0-4f490d0e2cd6" />